stl/inc/yvals_core.h

```diff
@@ -417,7 +423,25 @@
 #include <vcruntime.h>
 #include <xkeycheck.h> // The _HAS_CXX tags must be defined before including this.
 
-#define _WARNING_MESSAGE(NUMBER, MESSAGE) __FILE__ "(" _CRT_STRINGIZE(__LINE__) "): warning " NUMBER ": " MESSAGE
+// Note that _STL_PRAGMA is load-bearing;
+// it still needs to exist even once CUDA and ICC support _Pragma.
+#if defined(__CUDACC__) || defined(__INTEL_COMPILER)
+#define _STL_PRAGMA(PRAGMA) __pragma(#PRAGMA)

```

Comment:
This is incorrect because Standard `_Pragma` takes a string, while classic `__pragma` takes raw tokens:

https://github.com/microsoft/STL/blob/b7e0365b5e2f9de738e8fbea42b6cc6880efe531/stl/inc/yvals_core.h#L590-L602

This can be verified with MSVC itself (as checking CUDA/Intel is logistically difficult), by commenting out the conditional logic and always using classic `__pragma`. In that case, this happens:

```
D:\GitHub\STL>type meow.cpp
#include <charconv>
int main() {}

D:\GitHub\STL>cl /EHsc /nologo /W4 /std:c++14 meow.cpp
meow.cpp
D:\GitHub\STL\out\build\x64\out\inc\charconv(13): warning C4081: expected 'identifier'; found 'string'
```

Removing the stringizing for classic `__pragma` fixes the warning message:

```
D:\GitHub\STL>cl /EHsc /nologo /W4 /std:c++14 meow.cpp
meow.cpp
D:\GitHub\STL\out\build\x64\out\inc\charconv(13): warning STL4038: The contents of <charconv> are available only with C++17 or later.
```